kernel/src/hil/flash.rs

```diff
@@ -0,0 +1,24 @@
+pub enum Error {
+}
+
+pub type FlashResult = Result<(), Error>;
+
+/// A block of readable persistent flash memory.
+pub trait FlashReadable {
+    type Word;
+
+    /// Gets the size of the flash memory in words.
+    fn len(&self) -> usize;
+
+    /// Reads into a buffer from flash memory.
+    fn read(&self, offset: usize, buf: &mut [Self::Word]) -> FlashResult;
```

Comment:
Interesting that you need this. At least on the SAM4L I believe you can just read the memory directly.